content/zh/docs/tasks/configure-pod-container/attach-handler-lifecycle-event.md

```diff
@@ -0,0 +1,175 @@
+-->
+在本练习中，你将创建一个带有某个容器的 Pod，该容器有对应的钩子函数响应 postStart 和 preStop 事件
+
+<!--
+Here is the configuration file for the Pod:
+-->
+下面是对应 Pod 的配置文件
+
+{{< codenew file="pods/lifecycle-events.yaml" >}}
+
+<!--
+In the configuration file, you can see that the postStart command writes a `message`
+file to the Container's `/usr/share` directory. The preStop command shuts down
+nginx gracefully. This is helpful if the Container is being terminated because of a failure.
+-->
+在上述配置文件中，你可以看到 postStart 命令写了相应的信息到容器的 `/usr/share` 目录下的
+`message` 文件中，而 preStop 优雅的退出了 nginx。当程序发声错误时，上述命令对于容器的退
+出将十分有用。
+
+<!--
+Create the Pod:
+-->
+创建 Pod：
+
+    kubectl apply -f https://k8s.io/examples/pods/lifecycle-events.yaml
+
+<!--
+Verify that the Container in the Pod is running:
+-->
+验证 Pod 中的容器已经运行：
+
+    kubectl get pod lifecycle-demo
+
+<!--
+Get a shell into the Container running in your Pod:
+-->
+使用 shell 连接到你的 Pod 里的容器：
+
+    kubectl exec -it lifecycle-demo -- /bin/bash
+
+<!--
+In your shell, verify that the `postStart` handler created the `message` file:
+-->
+在 shell 中，验证 `postStart` 钩子创建了 `message` 文件：
+
+    root@lifecycle-demo:/# cat /usr/share/message
+
+<!--
+The output shows the text written by the postStart handler:
+-->
+命令行将输出 `postStart` 钩子写入的内容
+
+    Hello from the postStart handler
+
+{{% /capture %}}
+
+
+
+{{% capture discussion %}}
+
+<!--
+## Discussion
+-->
+## 讨论
+
+<!--
+Kubernetes sends the postStart event immediately after the Container is created.
+There is no guarantee, however, that the postStart handler is called before
+the Container's entrypoint is called. The postStart handler runs asynchronously
+relative to the Container's code, but Kubernetes' management of the container
+blocks until the postStart handler completes. The Container's status is not
+set to RUNNING until the postStart handler completes.
+-->
+Kubernetes 在容器创建后立即发送 postStart 事件。然而，postStart 钩子函数不保证运行在容器
+的控件挂载点(entrypoint)之前。postStart 钩子函数与容器的代码是异步化执行的，但 Kubernetes
+在 postStart 钩子执行完毕前将一直阻塞容器。直到 postStart 钩子函数执行完毕，容器的状态才会变成
+运行中(RUNNING)
+
+<!--
+Kubernetes sends the preStop event immediately before the Container is terminated.
+Kubernetes' management of the Container blocks until the preStop handler completes,
+unless the Pod's grace period expires. For more details, see
+[Termination of Pods](/docs/user-guide/pods/#termination-of-pods).
+-->
+Kubernetes 在容器结束前立即发送 preStop 事件。除非 Pod 优雅阶段超时，Kubernetes 在
+preStop 钩子执行完毕前将一直阻塞容器。更多的细节可以在
+[Pods 的结束](/docs/user-guide/pods/#termination-of-pods)中了解。
+
+<!--
+{{< note >}}
+Kubernetes only sends the preStop event when a Pod is *terminated*.
+This means that the preStop hook is not invoked when the Pod is *completed*. 
+This limitation is tracked in [issue #55087](https://github.com/kubernetes/kubernetes/issues/55807).
+{{< /note >}}
+-->
+{{< note >}}
+Kubernetes 只有在 Pod *结束*的时候才会发送 preStop 事件，这意味着在 Pod *完成* 时
+preStop 的钩子将不会被触发。这个限制在
+[issue #55087](https://github.com/kubernetes/kubernetes/issues/55807) 中被追踪。
```

Comment:
Is it worth localizing `issue`?

Reply:
Because we use `issure` as a proper noun in GitHub in our daily communication, I quote the original word here.